services/autoscaling/src/simcore_service_autoscaling/utils/utils_docker.py

```diff
@@ -646,6 +650,35 @@ async def get_node_empty_since(node: Node) -> datetime.datetime | None:
     )  # mypy
 
 
+async def set_node_begin_termination_process(
+    docker_client: AutoscalingDocker, node: Node
+) -> Node:
+    """sets the node to drain and adds a docker label with the time"""
+    assert node.Spec  # nosec
+    new_tags = deepcopy(cast(dict[DockerLabelKey, str], node.Spec.Labels))
+    new_tags[_OSPARC_NODE_TERMINATION_PROCESS_LABEL_KEY] = arrow.utcnow().isoformat()
+
+    return await tag_node(
+        docker_client,
+        node,
+        tags=new_tags,
+        available=False,
+    )
+
+
+def get_node_termination_started_since(node: Node) -> datetime.datetime | None:
+    assert node.Spec  # nosec
+    assert node.Spec.Labels  # nosec
```

Comment:
Suggestion:
```python
if dt := node.Spec.Labels.get(_OSPARC_NODE_TERMINATION_PROCESS_LABEL_KEY):
 return arrow.get(dt).datetime
return None
```

Reply:
yep, nice one. will try to remember that for the next one.